tests-beta/android/MASVS-CODE/MASTG-TEST-0223.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Stack Canaries Not Enabled
+platform: android
+id: MASTG-TEST-0223
+type: [static]
+weakness: MASWE-0116
+---
+
+## Overview
+
+This test case checks if the [native libraries](../../../Document/0x05i-Testing-Code-Quality-and-Build-Settings/#binary-protection-mechanisms) of the app are compiled without stack canaries and therefore lacking [stack smashing protection](../../../Document/0x04h-Testing-Code-Quality/#stack-smashing-protection), a common mitigation technique against buffer overflow attacks.
+
+- NDK libraries should have stack canaries enabled since [the compiler does it by default](https://android.googlesource.com/platform/ndk/%2B/master/docs/BuildSystemMaintainers.md#additional-required-arguments).
+- Other custom C/C++ libraries might not have stack canaries enabled because they lack the necessary compiler flags (`-fstack-protector-strong`, or `-fstack-protector-all`) or the libraries were optimized out by the compiler. See the [Evaluation](#evaluation) section for more details.
+
+## Steps
+
+1. Extract the app contents (@MASTG-TECH-0058).
+2. Run @MASTG-TECH-0115 on each shared library and grep for "canary" or the corresponding keyword used by the selected tool.
+
+## Observation
+
+The output should show if stack canaries are enabled or disabled.
+
+## Evaluation
+
+The test case fails if stack canaries are disabled.
+
+Developers need to ensure that the flags `-fstack-protector-strong`, or `-fstack-protector-all` are set in the compiler flags for all native libraries. This is especially important for custom C/C++ libraries that are not part of the NDK.
+
+When evaluating this please note that there are potential **expected false positives** and the test case should be considered as passed. But this would require a manual review of the original source code and the compilation flags used. The following examples cover some of the false positive cases that might be encountered:
```

Comment:
```suggestion
When evaluating this please note that there are potential **expected false positives** for which the test case should be considered as passed. To be certain for these cases, they require manual review of the original source code and the compilation flags used. 

The following examples cover some of the false positive cases that might be encountered:
```

Not sure about the " for which the test case should be considered as passed". For the first example below, I agree, but for the second I think this is dangerous.